pkgs/development/libraries/flutter-engine/default.nix

```diff
@@ -0,0 +1,399 @@
+            mkdir -p $out/src/third_party/dart/.dart_tool
+            cp -r --no-preserve=ownership,mode $dartPackageConfig $out/src/third_party/dart/.dart_tool/package_config.json
+
+            # Wrap clang
+            wrapProgram $out/src/buildtools/linux-x64/clang/bin/clang \
+              --add-flags "-isysroot ${toolchain}"
+
+            wrapProgram $out/src/buildtools/linux-x64/clang/bin/ld.lld \
+              --add-flags "-sysroot ${toolchain}"
+
+            # Run gclient's hooks
+            python3 $out/src/third_party/dart/tools/generate_package_config.py
+            python3 $out/src/third_party/dart/tools/generate_sdk_version_file.py
+            python3 $out/src/tools/remove_stale_pyc_files.py src/tools
+            python3 $out/src/flutter/tools/pub_get_offline.py
+            ${optionalString (targetOS == "wasm") "python3 src/flutter/tools/activate_emsdk.py"}
+          '';
+        };
+      in
+      stdenvNoCC.mkDerivation rec {
+        pname = "flutter-engine-${runtimeMode}";
+        version = builtins.substring 0 7 args.version;
+        inherit buildInputs enableParallelBuilding runtimeMode src toolchain configureFlags;
+
+        nativeBuildInputs = [ git python3 ninja stdenv.cc.libc ];
+
+        extraTargets = (args.extraTargets or [])
+          ++ (optional buildGtk "flutter_gtk");
+
+        passthru = {
+          inherit buildGtk enableFontconfig;
+        };
+
+        requiredSystemFeatures = optional enableParallelBuilding "big-parallel";
+
+        configurePhase = ''
+          # Update time
+          find src -type f -exec touch {} +
+
+          # Generate Ninja
+          export PKG_CONFIG_PATH=$toolchain/usr/lib/pkgconfig
+          export PKG_CONFIG_FOR_TARGET=$toolchain/usr/bin/pkg-config
+          export PKG_CONFIG_PATH_FOR_TARGET=$toolchain/usr/lib/pkgconfig
+          export PATH=$toolchain/usr/bin:$PATH
+
+          ./src/flutter/tools/gn \
+            --runtime-mode $runtimeMode \
+            --no-dart-version-git-info \
+            --no-prebuilt-dart-sdk \
+            --no-goma \
+            --clang \
+            --embedder-for-target \
+            --out-dir $out/lib/flutter \
+            --target-sysroot $toolchain \
+            --target-triple ${targetPlatform.config} \
+            --target-dir $runtimeMode \
+            --target-os ${targetOS} \
+            ${optionalString enableFontconfig "--enable-fontconfig"} \
+            ${optionalString (targetOS != "wasm") "--${targetOS}-cpu ${targetCPU}"} \
+            ''${configureFlags[@]}
+        '';
+
+        buildPhase = ''
+          cd $out/lib/flutter/out/$runtimeMode
+          export TERM=dumb
+
+          # Comes from "pkgs/development/tools/build-managers/ninja/setup-hook.sh"
+
+          local buildCores=1
+
+          if [ "''${enableParallelBuilding-1}" ]; then
+            buildCores="$NIX_BUILD_CORES"
+          fi
+
+          local flagsArray=(
+            -j$buildCores
```

Comment:
add `-l$buildCores` to limit cpu load
this is actually a bug in pkgs/development/tools/build-managers/ninja/setup-hook.sh

Reply:
According to #192447, that can cause issues. I don't want to change it until we know the `-j` actually causes more issues.